packages/hash/frontend/src/components/hooks/useCreateComment.ts

```diff
@@ -0,0 +1,31 @@
+import { useMutation } from "@apollo/client";
+import { TextToken } from "@hashintel/hash-shared/graphql/types";
+import { useCallback } from "react";
+import {
+  CreateCommentMutation,
+  CreateCommentMutationVariables,
+} from "../../graphql/apiTypes.gen";
+import { createComment } from "../../graphql/queries/comment.queries";
+
+export const useCreateComment = (
+  accountId: string,
+): [
+  (parentId: string, content: TextToken[]) => Promise<void>,
+  { loading: boolean },
+] => {
+  const [createCommentFn, { loading }] = useMutation<
+    CreateCommentMutation,
+    CreateCommentMutationVariables
+  >(createComment);
+
+  const createBlockComment = useCallback(
+    async (parentId: string, content: TextToken[]) => {
+      await createCommentFn({
+        variables: { accountId, parentId, content },
+      });
+    },
+    [createCommentFn, accountId],
+  );
+
+  return [createBlockComment, { loading }];
+};
```

Comment:
You don't need to manually type this 

```suggestion
) => {
  const [createCommentFn, { loading }] = useMutation<
    CreateCommentMutation,
    CreateCommentMutationVariables
  >(createComment);

  const createBlockComment = useCallback(
    async (parentId: string, content: TextToken[]) => {
      await createCommentFn({
        variables: { accountId, parentId, content },
      });
    },
    [createCommentFn, accountId],
  );

  return [createBlockComment, { loading }] as const;
};
```

Reply:
Thank you, made this change in 85e264142874e87a9f1c13a77c51197727f39da4 and also went back and changed some other files where I followed the same pattern (in #931).